datafusion/physical-expr/src/aggregate/build_in.rs

```diff
@@ -417,7 +417,18 @@ pub fn create_aggregate_expr(
         (AggregateFunction::StringAgg, true) => {
             return not_impl_err!("STRING_AGG(DISTINCT) aggregations are not available");
         }
-    })
+    };
+
+    let agg_name = aggregate_expr.name();
+    if ignore_nulls && !aggregate_expr.support_ignore_nulls() {
```

Comment:
I would prefer these checks at the beginning of the function, so we can avoid unnecessary compute for invalid cases